pkg/cloud/amazon/vault/vault_resources.go

```diff
@@ -0,0 +1,342 @@
+package vault
+
+import (
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/aws/aws-sdk-go/service/iam"
+	goformation "github.com/awslabs/goformation/cloudformation"
+	"github.com/awslabs/goformation/cloudformation/resources"
+	"github.com/jenkins-x/jx/pkg/log"
+	"github.com/jenkins-x/jx/pkg/util"
+)
+
+const stackName = "jenkins-cloud-formation-gitops-stack"
+
+// CreateVaultResources will automatically create the preresiquites for vault on aws
+func CreateVaultResources(region string, domain string, username string, tableName string) (*string, *string, *string, *string, error) {
+	log.Infof("Creating vault presiquite resources with following values, %s, %s, %s, %s",
+		util.ColorInfo(region),
+		util.ColorInfo(domain),
+		util.ColorInfo(username),
+		util.ColorInfo(tableName))
+
+	template := goformation.NewTemplate()
+
+	awsDynamoDbKey := "AWSDynamoDBTable"
+	dynamoDbTable := createDynamoDbTable(tableName)
+	template.Resources[awsDynamoDbKey] = &dynamoDbTable
+
+	awsS3BucketKey := "AWSS3Bucket"
+	s3Name, s3Bucket := createS3Bucket(region, domain)
```

Comment:
Are these names unique per cluster?

Reply:
It will conflict if domain name is same, I will add a random hash